velox/exec/Operator.cpp

```diff
@@ -115,10 +115,32 @@ void Operator::maybeSetTracer() {
     return;
   }
 
-  if (queryTraceConfig->queryNodes.count(planNodeId()) == 0) {
+  const auto nodeId = planNodeId();
+  if (queryTraceConfig->queryNodes.count(nodeId) == 0) {
     return;
   }
 
+  {
+    const auto& tracedNodeMap =
+        operatorCtx_->driverCtx()->tracedNodeMap_.rlock();
```

Comment:
This probably won't work for hash join. Could we extend query node map to have operator types in queryTraceConfig?

Reply:
The `queryTraceConfig` is shared by all operators in the query, and there will be multiple operator instances for the same node ID. The `HashBuild` and `HashProbe` operators created from the same HashJoinNode belong to different pipelines and they will create different tracing data dirs.